libtransmission/blocklist.cc

```diff
@@ -71,133 +54,109 @@ static void blocklistLoad(tr_blocklistFile* b)
     }
 
     tr_error* error = nullptr;
-    auto const fd = tr_sys_file_open(b->filename, TR_SYS_FILE_READ, 0, &error);
+    auto const fd = tr_sys_file_open(getFilename(), TR_SYS_FILE_READ, 0, &error);
     if (fd == TR_BAD_SYS_FILE)
     {
         tr_logAddWarn(fmt::format(
             _("Couldn't read '{path}': {error} ({error_code})"),
-            fmt::arg("path", b->filename),
+            fmt::arg("path", getFilename()),
             fmt::arg("error", error->message),
             fmt::arg("error_code", error->code)));
         tr_error_free(error);
         return;
     }
 
-    b->rules = static_cast<struct tr_ipv4_range*>(tr_sys_file_map_for_reading(fd, 0, byteCount, &error));
-    if (b->rules == nullptr)
+    rules_ = static_cast<struct IPv4Range*>(tr_sys_file_map_for_reading(fd, 0, byteCount, &error));
+    if (rules_ == nullptr)
     {
         tr_logAddWarn(fmt::format(
             _("Couldn't read '{path}': {error} ({error_code})"),
-            fmt::arg("path", b->filename),
+            fmt::arg("path", getFilename()),
             fmt::arg("error", error->message),
             fmt::arg("error_code", error->code)));
         tr_sys_file_close(fd);
         tr_error_free(error);
         return;
     }
 
-    b->fd = fd;
-    b->byteCount = byteCount;
-    b->ruleCount = byteCount / sizeof(struct tr_ipv4_range);
+    fd_ = fd;
+    byte_count_ = byteCount;
+    rule_count_ = byteCount / sizeof(IPv4Range);
 
     tr_logAddInfo(fmt::format(
-        ngettext("Blocklist '{path}' has {count} entry", "Blocklist '{path}' has {count} entries", b->ruleCount),
-        fmt::arg("path", tr_sys_path_basename(b->filename)),
-        fmt::arg("count", b->ruleCount)));
+        ngettext("Blocklist '{path}' has {count} entry", "Blocklist '{path}' has {count} entries", rule_count_),
+        fmt::arg("path", tr_sys_path_basename(getFilename())),
+        fmt::arg("count", rule_count_)));
 }
 
-static void blocklistEnsureLoaded(tr_blocklistFile* b)
+void BlocklistFile::ensureLoaded()
 {
-    if (b->rules == nullptr)
+    if (rules_ == nullptr)
     {
-        blocklistLoad(b);
+        load();
     }
 }
 
-static int compareAddressToRange(void const* va, void const* vb)
-{
-    auto const* a = static_cast<uint32_t const*>(va);
-    auto const* b = static_cast<struct tr_ipv4_range const*>(vb);
-
-    if (*a < b->begin)
-    {
-        return -1;
-    }
-
-    if (*a > b->end)
-    {
-        return 1;
-    }
-
-    return 0;
-}
-
-static void blocklistDelete(tr_blocklistFile* b)
-{
-    blocklistClose(b);
-    tr_sys_path_remove(b->filename);
-}
+// TODO: unused
+//static void blocklistDelete(tr_blocklistFile* b)
+//{
+//    blocklistClose(b);
+//    tr_sys_path_remove(b->filename, nullptr);
+//}
```

Comment:
It looks like this is an intentional decision to not delete blocklists?

From the existing codepaths for `tr_blocklistFileSetContent()` I think that's probably a good call. Looks like this code was started with a feature in mind that never came together; it's not really clear to me how it ever got invoked.

What did you have in mind with this `TODO` comment?

Reply:
When refactoring i check usages and comment dead stuff, if it builds and tests with dead stuff out, then its out.